pkg/cmd/kafka/topic/consume/consume.go

```diff
@@ -0,0 +1,182 @@
+	flags.StringVar(&opts.timestamp, "timestamp", "", f.Localizer.MustLocalize("kafka.topic.consume.flag.timestamp.description"))
+	flags.BoolVar(&opts.wait, "wait", false, f.Localizer.MustLocalize("kafka.topic.consume.flag.wait.description"))
+	flags.Int64Var(&opts.offset, "offset", 0, f.Localizer.MustLocalize("kafka.topic.consume.flag.offset.description"))
+	flags.Int32Var(&opts.limit, "limit", 20, f.Localizer.MustLocalize("kafka.topic.consume.flag.limit.description"))
+
+	_ = cmd.MarkFlagRequired("name")
+
+	_ = cmd.RegisterFlagCompletionFunc("name", func(cmd *cobra.Command, _ []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return kafkacmdutil.FilterValidTopicNameArgs(f, toComplete)
+	})
+
+	flags.AddInstanceID(&opts.kafkaID)
+
+	return cmd
+}
+
+func runCmd(opts *options) error {
+
+	conn, err := opts.f.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api, _, err := conn.API().KafkaAdmin(opts.kafkaID)
+	if err != nil {
+		return err
+	}
+
+	if opts.wait {
+		
+		max_offset := opts.offset
+		for true {
+
+			records, err := consume(opts, api);	
+			if err != nil {
+				return err
+			}
+
+			record_count := len(records.Items) 
+			if record_count > 0 {
+				max_offset = *(records.Items[record_count - 1].Offset) + 1
+				outputRecords(opts, records)
+			}	
+
+			opts.offset = max_offset 
+		}
+	} else {
+		records, err := consume(opts, api)
+		if err != nil {
+			return err
+		}
+
+		outputRecords(opts, records)
+	}
+
+	return nil
+}
+
+func consume(opts *options, api *kafkainstanceclient.APIClient) (*kafkainstanceclient.RecordList, error) {
+
+	request := api.RecordsApi.ConsumeRecords(opts.f.Context, opts.topicName).Limit(opts.limit).Partition(opts.partition).Offset(int32(opts.offset))
+	if opts.timestamp != "" {
+		// setting timestamp as "" (not set by user) is not valid
+		// not setting timestamp is handled by the request
+		request = request.Timestamp(opts.timestamp)
+	}
+
+	list, _, err := request.Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	return &list, nil
+}
+
+func outputRecords(opts *options, records *kafkainstanceclient.RecordList) {
+	recordsAsRows := mapRecordsToRows(opts.topicName, &records.Items)
+
+	if len(records.Items) == 0 {
+		opts.f.Logger.Info(opts.f.Localizer.MustLocalize("kafka.common.log.info.noRecords"))
+		return
+	}
+
+	switch opts.outputFormat {
+	case dump.EmptyFormat:
+		dump.Table(opts.f.IOStreams.Out, recordsAsRows)
+		opts.f.Logger.Info("")
+	default:
+		dump.Formatted(opts.f.IOStreams.Out, opts.outputFormat, recordsAsRows)
+		opts.f.Logger.Info("")
+	}
+}
+
+func mapRecordsToRows(topic string, records *[]kafkainstanceclient.Record) []kafkaRow {
+
+	rows := make([]kafkaRow, len(*records))
+
+	for i := 0; i < len(*records); i++ {
+		record := &(*records)[i]
+		row := kafkaRow{
+			Topic:     topic,
+			Key:       *record.Key,
+			Value:     strings.TrimSuffix(record.Value, "\n"), // trailing new line gives weird printing of table
```

Comment:
Value in table format would be very impractical.
Have tried it with 2 examples from here:

https://support.oneskyapp.com/hc/en-us/articles/208047697-JSON-sample-files

Even small multiline json breaks table right now

Reply:
UI is using table because they have capabilities to view more detail on click

Reply:
Btw. I still think table is good but not sure what to do with value and key properties that are almost always long

Reply:
Jut going to set default to json output